### PR TITLE
Ansible py3

### PR DESCRIPTION
**Pull request by sengaya**
###### Motivation for this change

- Switch to python3 for ansible and related packages
- Remove ansible 2.4 as it [does not support python 3](https://github.com/NixOS/nixpkgs/pull/40277#discussion_r187281450) and is already [out of support](https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#id14)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

This PR should make #44413 obsolete.
closes #44413

I guess I need to provide an update to the release notes for 19.03, but I would first like to get some feedback it this change has the chance to get merged.

cc @jgeerds @joamaki @c0bw3b

### CODE REVIEW

**Comment by c0bw3b** [comment]
+1 for the removal of `ansible_2_4` since it has unpatched CVEs (probably).

@GrahamcOfBorg build ansible_2_7 ansible-lint

**Comment by c0bw3b** [comment]
At least it builds successfully with Python 3.6

**Comment by FRidh** [comment]
Note on staging-next we've switched to `python3 = python37`.

File: pkgs/development/tools/ansible-lint/default.nix [hunk 1]
```diff
@@ -1,6 +1,6 @@
-{ stdenv, fetchFromGitHub, pythonPackages, ansible }:
+{ stdenv, fetchFromGitHub, python3Packages, ansible }:
```
Comment (dotlambda) [inline pkgs/development/tools/ansible-lint/default.nix]: You might want to replace `python3Packages` by `python3` ...

File: pkgs/development/tools/ansible-lint/default.nix [hunk 2]
```diff
@@ -11,9 +11,9 @@ pythonPackages.buildPythonPackage rec {
     sha256 = "0cnfgxh5m7alzm811hc95jigbca5vc1pf8fjazmsakmhdjyfbpb7";
   };
 
-  propagatedBuildInputs = with pythonPackages; [ pyyaml six ] ++ [ ansible ];
+  propagatedBuildInputs = with python3Packages; [ pyyaml six ] ++ [ ansible ];
```
Comment (dotlambda) [inline pkgs/development/tools/ansible-lint/default.nix]: ... and `ansible` by `ansible.override { python3 = python3; }`.
That way, both are using the same set of python packages.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by Mic92** [comment]
cc @Lassulus

**Comment by Mic92** [comment]
I guess mentioning the python3 switch is worth a changelog line.
Dropping ansible 2.4 is something to be expected since we regularly remove unsupported versions.

**Comment by costrouc** [comment]
Why are `ansible` and `ansible-list` not moved into `python-modules`? Two packages `pytest-ansible` and `ansible-lint` already depend on `ansible` as a module. I am working on two packages `ansible-kernel` and `ansible-runner` that both have python 27 and 3.5+ support and depend on `ansible` as a module.

**Comment by dotlambda** [comment]
> Why are `ansible` and `ansible-list` not moved into `python-modules` and exposed as applications in `all-packages` for compatibility?

Sounds like a sane thing to do.

**Comment by costrouc** [comment]
Added a note here that one of my PRs that depends on this one moving to `python-modules`. There is a jupyter kernel for ansible that has support for python 2.7 and 3+. @sengaya let me know if you need any help!

**Comment by c0bw3b** [comment]
`ansible` and `ansible-lint` are applications, first and foremost.
They should stay visible in `all-packaging.nix`

**Comment by dotlambda** [comment]
> `ansible` and `ansible-lint` are applications, first and foremost.
> They should stay visible in `all-packages.nix`

That's absolutely possible by adding e.g.
```
anisble = with python3.pkgs; toPythonApplication ansible;
```
to `all-packages.nix`.

**Comment by dasJ** [comment]
Bumping this up because I really need it (currently pinning ansible manually to use python 3).

**Pull request closed without merging**